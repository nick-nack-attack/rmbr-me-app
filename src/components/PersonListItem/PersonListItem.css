.PersonListItem__heading {
    text-align: left;
    margin: 0;
}

.PersonListItem__header, .person_category {
    padding: 0;
}

.person_category {
    text-align: left;
    margin: 0;
}

.PersonListItem__footer {
    padding: 0;
    width: 20px;
}

.personListItem__button {
    position: relative;
    margin-left: 0;
}

.personListItem__div_button {
    width: 150px;
    text-align: left;
}