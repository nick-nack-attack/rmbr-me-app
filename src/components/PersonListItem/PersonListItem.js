import React, {Component} from 'react';
import RmbrmeContext from "../../contexts/RmbrmeContext";
import RmbrApiService from "../../services/rmbr-api-service";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';


export default class PersonListItem extends Component {

    constructor(props) {
        super(props);
        this.state = {
            error: null,
            handleClickPerson: () => {}
        };
    };

    static contextType = RmbrmeContext;

    handleClickPerson = (person_id) => {
        this.props.handleClickPerson(person_id);
    }

    handleDeletePerson = (e, personId) => {
        e.stopPropagation();
        this.setState({error: null});
        RmbrApiService.deletePerson(personId)
            .then(this.context.deletePerson(personId))
            .catch(res => {
                this.setState({
                    error: res.error
                });
            })
    };

    render() {
        const person_id = this.props.person.id;
        const makeARmbrPromptButton = <button onClick={() => this.handleClickPerson(person_id)}> <FontAwesomeIcon icon='plus' /> Create a Rmbr </button>;
        const arrayLength = this.props.rmbrArray.length;
        const PersonRmbrList = !arrayLength
            ? makeARmbrPromptButton
            : (this.props.rmbrArray.find(rbr => rbr.person_id === this.props.person.id)).rmbr_title
        const rmbrOrRmbrs = arrayLength === 1 ? 'rmbr' : 'rmbrs';

    return (
        <>
            <li
                key={person_id}
                className='PersonListItem'
                onClick={() => this.handleClickPerson(person_id)}
            >
                <button
                    className='list_delete_button person_list_delete-button'
                    onClick={(e) => this.handleDeletePerson(e, this.props.person.id)}
                >
                    <FontAwesomeIcon icon='trash-alt' />
                </button>
                <h3
                    className='PersonListItem__heading'
                >
                    { this.props.person.person_name }
                </h3>
                <p className='person_category'>{ this.props.person.type_of_person } <FontAwesomeIcon icon='bolt'/> {this.props.rmbrArray.length} {rmbrOrRmbrs}</p>
                <div
                    className='latest_rmbr'
                >
                    { !arrayLength ? `No Rmbrs Yet!` : `` }
                    <div>
                        { PersonRmbrList }
                    </div>
                </div>
            </li>
        </>
        );
    };
};