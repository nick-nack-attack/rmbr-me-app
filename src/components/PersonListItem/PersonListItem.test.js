import React, { Component } from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PersonListItem from "./PersonListItem";

const context = React.createContext({
    rmbrArray: []
});

export class testContext extends Component {
    state = {
        rmbrArray: ['One item', 'Two items']
    }
    render() {
        const value = {
            rmbrArray: this.state.rmbrArray
        }
        return (
            <testContext.Provider value={value}>
                { this.props.children }
            </testContext.Provider>
        );
    };
};

describe(`PersonListItem component`, () => {

    it('renders PersonListItem by default', () => {
        const wrapper = shallow(
        <testContext>
            <PersonListItem context={context}/>
        </testContext>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})