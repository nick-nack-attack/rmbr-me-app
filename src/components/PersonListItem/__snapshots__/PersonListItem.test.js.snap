// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonListItem component renders PersonListItem by default 1`] = `
<testContext>
  <PersonListItem
    context={
      Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "rmbrArray": Array [],
        },
        "_currentValue2": Object {
          "rmbrArray": Array [],
        },
        "_threadCount": 0,
      }
    }
  />
</testContext>
`;
